text/0002-bundle-add-and-remove.md

```diff
@@ -0,0 +1,275 @@
+- Feature Name: bundle\_add\_and\_remove\_improvements
+- Start Date: 2018-02-10
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+This feature adds improvements for `bundle add` command and adds
+the opposite command for removing gems called `bundle remove`.
+
+The purpose of this RFC is to get feedback on use cases, CLI format/syntax
+and feature options.
+
+# Motivation
+
+The main use case is improving user productivity. 
+
+This is done by adding gem(s) to the Gemfile with the latest version in the range specified from the command line. User saves time since they don't have to search Rubygems for the latest version. 
+
+The opposite command is also available, which provides convenience for removing gem(s) from the Gemfile from the command line.
+
+# Guide-level explanation
+
+The usage of the `add` and `remove` command can be grouped in 7 sections.
+
+## 1. adding gem to group block
+
+### 1.1. when Gemfile doesn't yet have the group specified
+
+#### Proposed command
+
+```ruby
+# Gemfile before
+source "https://rubygems.org"
+```
+
+```ruby
+bundle add minitest --group test
+```
+
+```ruby
+# Gemfile after
+source "https://rubygems.org"
+
+group :test do
+  gem "minitest", "~> 5.11"
```

Comment:
I don't agree with adding a version constraint when it wasn't specified by the user. The version constraint should be defined  by the `--version` option that `bundle add` has already.

Reply:
`bundle add` already works this way if you don't specify the version. How should it behave if the version is not specified?

Reply:
ah ok, didn't know that. nevermind then

Reply:
@colby-swandale FYI, this was @indirect default requirement in the original `add` command. He wanted conservative versioning to be set as it's the safest and can be overridden if the user added a specific version.